### PR TITLE
Initial plugin setup

### PR DESCRIPTION
**Pull request by joshuapease**
### Overview

This is the first working version of a standalone Phone Home plugin. 
- Extracts the Phone Home feature from our Viget Base Module
- Uses Notion instead of Airtable

### Why are we doing this?

Previously this functionality was tied to our Viget Base module. But a few things prompted the idea of breaking this into it's own plugin.

- Airtable's API was changing, and we use Notion more these days.
- Having a third-party API tied to our base module felt a little brittle. 
    - Forcing us to make updates when we didn't want to.
- A smaller simpler plugin would be easier to share with the broader Craft community.

### CODE REVIEW

File: src/config.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    // An array of classes that implement the \viget\phonehome\endpoints\EndpointInterface
+    // You may configure as many endpoints as you would like... even of the same class
+    // 'endpoints' => [
+    //     new \viget\phonehome\endpoints\NotionEndpoint(
+    //         secret: \craft\helpers\App::env('PHONE_HOME_NOTION_SECRET'),
+    //         databaseId: \craft\helpers\App::env('PHONE_HOME_NOTION_DATABASE'),
+    //     ),
+    // ],
+];
```
Comment (joshuapease) [inline src/config.php:24]: This makes it easy to configure multiple endpoints if the need ever arises. 

I could see this being useful with a client who has multiple Craft sites. 

We could configure one Notion endpoint for our Viget database. But also set up the client with their own database of just their sites.

**Review by maxdmyers — APPROVED** [review]

**Comment by maxdmyers** [comment]
Nice work on this, @joshuapease!

**Review by JJimmyFlynn — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -28,3 +28,34 @@ composer require viget/craft-phone-home
...
+
+Follow the Notion docs to create an [internal integration](https://developers.notion.com/docs/getting-started#internal-integrations). You'll need to be a Workspace owner or use your personal workspace to set up an integration.
+
+At the end of this process, you'll have a Notion integration secret key. You'll need this to configure the plugin.
+
+### 2. Create a database and add it to your Notion Integration
+
+Create a [new database](https://www.notion.so/help/guides/creating-a-database) in Notion. 
+
+No need to add any columns or properties. This plugin will create them when it's first run.
```
Comment (ten1seven) [inline README.md:48]: 😍

File: src/jobs/SendPayloadJob.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+class SendPayloadJob extends BaseJob
+{
+    function execute($queue): void
+    {
+        PhoneHome::getInstance()->phoneHome->sendPayload();
+    }
+
+    protected function defaultDescription(): ?string
+    {
+        return 'Phoning Home';
```
Comment (ten1seven) [inline src/jobs/SendPayloadJob.php:20]: ☎️ Phoning Home
👽 Phoning Home

Options...

**Review by ten1seven — APPROVED** [review]

File: src/jobs/SendPayloadJob.php (hunk 3 above)
Comment (joshuapease) [reply src/jobs/SendPayloadJob.php:20]: That would be fun... I'm a little leery of emoji in SQL databases. It's probably not a huge issue... but I'd rather not think about it.

Should be resolved fully for Craft 5 (they're tweaking the default charset to be utf8mb4).

https://github.com/craftcms/cms/issues/12935#issuecomment-1495108327

**Review by maxfenton — COMMENTED** [review]
👍

**Pull request merged**